### PR TITLE
docs: Cleanup Python models section of README

### PR DESCRIPTION
**Pull request by dfsnow**
# Description

This PR adds a few of the limitations I found while trying to implement a Python model. It also cleans up some language in the docs and puts the sections in the correct order.

For reference, I have a Python model implementation here: https://github.com/ccao-data/data-architecture/pull/422.

## Checklist

- [x] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [x] You kept your Pull Request small and focused on a single feature or bug fix.
- [ ] You added unit testing when necessary
- [ ] You added functional testing when necessary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -277,7 +278,7 @@ athena:
       }
   ```
 
-> Notes:  
+> Notes:
```
Comment (dfsnow) [inline README.md:281]: This is just trailing whitespace cleanup performed by my editor.

File: README.md [hunk 2]
```diff
@@ -539,37 +540,143 @@ It's important to understand the following points:
...
+
+- Tables, schemas and database should only be lowercase
+
+- In order to avoid potential conflicts, make sure [`dbt-athena-adapter`](https://github.com/Tomme/dbt-athena) is not
+  installed in the target environment.
+  See <https://github.com/dbt-athena/dbt-athena/issues/103> for more details.
+
+- Snapshot does not support dropping columns from the source table. If you drop a column make sure to drop the column
+  from the snapshot as well. Another workaround is to NULL the column in the snapshot definition to preserve history
+
```
Comment (dfsnow) [inline README.md:656]: None of the content in this section has changed, it was just incorrectly ordered and has moved up from below the `Python Models` header.

**Comment by nicor88** [comment]
@dfsnow Could you please fix the linting issue on the markdown? 🙏🏻

**Comment by dfsnow** [comment]
> @dfsnow Could you please fix the linting issue on the markdown? 🙏🏻

@nicor88 Done!

File: README.md [hunk 3]
```diff
@@ -568,37 +569,143 @@ It's important to understand the following points:
...
-That's why you should handle this by yourself manually or using some automation tools like terraform, AWS CDK etc.  
+That's why you should handle this by yourself manually or using some automation tools like terraform, AWS CDK etc.
 You may find the following links useful to manage that:
 
 <!-- markdownlint-disable -->
 * [terraform aws_lakeformation_permissions](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lakeformation_permissions)
 * [terraform aws_lakeformation_resource_lf_tags](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lakeformation_resource_lf_tags)
 <!-- markdownlint-restore -->
 
+### Working example
```
Comment (nicor88) [inline README.md:580]: Working examples should refer to snapshots, but it's put after AWS Lakeformation section. Could you please fix it?

Comment (dfsnow) [reply README.md:557]: Done in 629da5e!

**Review by nicor88 — APPROVED** [review]

**Pull request merged**